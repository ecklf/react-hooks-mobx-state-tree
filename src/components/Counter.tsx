import { observer } from "mobx-react-lite";
import React from "react";
import { useMst } from "../models/Root";
import Button from "./Button";

const Counter = observer(() => {
  const { counter } = useMst();
  return (
    <div className="flex flex-col items-center mt-20">
      <p className="text-2xl font-bold text-center">Counter</p>
      <p
        style={{ fontVariant: "tabular-nums" }}
        className="text-2xl font-bold text-center"
      >
        {counter.count}
      </p>
      <div className="flex mt-2 space-x-3">
        <Button label="-" onClick={counter.decrement} />
        <Button label="+" onClick={counter.increment} />
      </div>
    </div>
  );
});

export default Counter;
