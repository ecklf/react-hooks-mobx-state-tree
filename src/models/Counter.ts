import { types } from "mobx-state-tree";

export const Counter = types
  .model({
    count: types.number
  })
  .actions(self => ({
    increment() {
      self.count++;
    },
    decrement() {
      self.count--;
    }
  }));
