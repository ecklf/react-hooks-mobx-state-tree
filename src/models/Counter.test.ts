import { Counter } from "./Counter";
import { getSnapshot, onPatch } from "mobx-state-tree";

it("should create a counter instance", () => {
  const item = Counter.create({
    count: 0
  });
  expect(getSnapshot(item)).toMatchSnapshot();
});

it("should increment and decrement", () => {
  const item = Counter.create({
    count: 0
  });
  const patches: any = [];

  onPatch(item, patch => {
    patches.push(patch);
  });

  item.increment();
  item.decrement();
  expect(patches).toMatchSnapshot();
});
