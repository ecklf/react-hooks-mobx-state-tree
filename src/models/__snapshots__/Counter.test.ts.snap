// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a counter instance 1`] = `
Object {
  "count": 0,
}
`;

exports[`should increment and decrement 1`] = `
Array [
  Object {
    "op": "replace",
    "path": "/count",
    "value": 1,
  },
  Object {
    "op": "replace",
    "path": "/count",
    "value": 0,
  },
]
`;
